### PR TITLE
Add end-user documentation for CCS using async-search

### LINKED ISSUE #73971: Async search support for cross cluster search minimize round trip optimization
[Async search](https://www.elastic.co/guide/en/elasticsearch/reference/current/async-search.html) does not currently support minimize roundtrip optimization when used with cross cluster search.  This is a documented limitation.  

With Kibana now using async searches by default, this could cause performance issues in some environments that use cross cluster search with slower/inconsistent network latency.  One workaround is to enable the deprecated advanced setting `courier:batchSearches` in Kibana which will switch Kibana back to using the old way of querying Elasticsearch via `msearch` (but this option is slated for removal in 8.0).

### PR DESCRIPTION
**Pull request by quux00**
End user docs have been updated, with new async-search based examples in the "modules-cross-cluster-search" page, which did not have any async-search examples previously.

### CODE REVIEW

**Comment by quux00** [comment]
@abdonpijpelink I believe this is ready for a docs review. This is my first time editing end user docs with examples that are tested, so let me know where I can improve.

File: docs/reference/search/search-your-data/search-across-clusters.asciidoc [hunk 1]
```diff
@@ -286,6 +286,321 @@ means the document came from the local cluster.
...
+
+
+
+
+[discrete]
+[[ccs-async-search-minimize-roundtrips-true]]
+=== Using async search for CCS with ccs_minimize_roundtrips=true
+
+The following <<async-search,async search>> API request searches the `my-index-000001` index on
+three clusters:
```
Comment (abdonpijpelink) [inline docs/reference/search/search-your-data/search-across-clusters.asciidoc:298]:
```suggestion
Remote clusters can be queried asynchronously using the 
<<async-search,async search>> API. Async searches accept a
<<ccs-minimize-roundtrips,`ccs_minimize_roundtrips`>> parameter that defaults to
`true`. 

The following request searches the `my-index-000001` index on three clusters 
asynchronously:
```

File: docs/reference/search/async-search.asciidoc [hunk 2]
```diff
@@ -123,9 +123,12 @@ number of new shard responses (`5` by default).
 enforce the execution of a pre-filter roundtrip to retrieve statistics from
 each shard so that the ones that surely don't hold any document matching the
 query get skipped.
-* `ccs_minimize_roundtrips` defaults to `false`. When doing a cross-cluster search,
-setting it to `true` may improve overall search latency, particularly when
-searching clusters with a large number of nodes.
+* `ccs_minimize_roundtrips` defaults to `true` (same as the synchronous search API).
+When doing a cross-cluster search, setting it to `true` may improve overall search
```
Comment (abdonpijpelink) [inline docs/reference/search/async-search.asciidoc:127]:
```suggestion
When doing a {ccs}, setting it to `true` may improve overall search
```

File: docs/reference/search/search-your-data/search-across-clusters.asciidoc (hunk 1 above)
Comment (abdonpijpelink) [reply docs/reference/search/search-your-data/search-across-clusters.asciidoc:298]: I think it's nice to introduce the Async API and the `ccs_minimize_roundtrips` parameter before jumping into the example.

**Review by abdonpijpelink — APPROVED** [review]
LGTM. I've left some minor suggestions.

Congrats on getting those code samples to pass the CI checks on the first attempt!

File: docs/reference/search/async-search.asciidoc (hunk 2 above)
Comment (quux00) [reply docs/reference/search/async-search.asciidoc:127]: Fixed. Thanks!

File: docs/reference/search/async-search.asciidoc [hunk 3]
```diff
@@ -123,9 +123,12 @@ number of new shard responses (`5` by default).
 enforce the execution of a pre-filter roundtrip to retrieve statistics from
 each shard so that the ones that surely don't hold any document matching the
 query get skipped.
-* `ccs_minimize_roundtrips` defaults to `false`. When doing a cross-cluster search,
-setting it to `true` may improve overall search latency, particularly when
-searching clusters with a large number of nodes.
+* `ccs_minimize_roundtrips` defaults to `false`. When doing a {ccs}, setting
+it to `true` may improve overall search latency, particularly when searching
+clusters with a large number of nodes. However, when set to `true`, search
```
Comment (javanna) [inline docs/reference/search/async-search.asciidoc:128]: I see number of nodes being mentioned, but like I mentioned before, the performance is a factor of the number of shards and not the number nodes.

**Review by javanna — APPROVED** [review]
LGTM besides the small comment I left

File: docs/reference/search/async-search.asciidoc (hunk 3 above)
Comment (javanna) [reply docs/reference/search/async-search.asciidoc:128]: ops, seems like I had started a review which I forgot to submit, no wonder why the comment was not addressed :)

Comment (quux00) [reply docs/reference/search/async-search.asciidoc]: Thanks. Fixed.

**Pull request merged**